api/odl.py

```diff
@@ -1518,3 +1608,41 @@ def _get(self, url, patron=None, headers=None, allowed_response_codes=None):
         self.request_args.append((patron, headers, allowed_response_codes))
         response = self.responses.pop()
         return HTTP._process_response(url, response, allowed_response_codes=allowed_response_codes)
+
+
+class ODLExpiredItemsReaper(IdentifierSweepMonitor):
+    """Responsible for removing expired ODL licenses."""
+
+    SERVICE_NAME = "ODL Expired Items Reaper"
+    PROTOCOL = ODLAPI.NAME
+
+    def __init__(self, _db, collection):
+        super(ODLExpiredItemsReaper, self).__init__(_db, collection)
+
+    def process_item(self, identifier):
+        for licensepool in identifier.licensed_through:
+            remaining_checkouts = 0   # total number of checkouts across all the licenses in the pool
+            concurrent_checkouts = 0  # number of concurrent checkouts allowed across all the licenses in the pool
+
+            # 0 is a starting point,
+            # we're going through all the valid licenses in the pool and count up available checkouts.
+            for license_pool_license in licensepool.licenses:
```

Comment:
This is the main addition to the second PR. Previously we were assuming that a LicensePool had only a single license.